imcsdk/apis/server/adaptor.py

```diff
@@ -13,61 +13,121 @@
 
 
 """
-This module provides apis to setup cisco vic adaptor properties and create vnics
-and vhbas
+This module provides apis to setup cisco vic adaptor properties \
+and create vnics and vhbas
 """
 
+from imcsdk.imccoreutils import get_server_dn
+from imcsdk.imcexception import ImcOperationError
 
-def get_vic_adaptor_properties(handle, adaptor_slot):
+
+def _is_valid_arg(param, kwargs):
+    return param in kwargs and kwargs[param] is not None
+
+
+def _check_and_get_param(param, kwargs):
+    if _is_valid_arg(param, kwargs):
+        return kwargs[param]
+    else:
+        return None
+
+
+def _set_server_dn(handle, kwargs):
+    if _is_valid_arg("server_id", kwargs):
+        server_id = str(kwargs["server_id"])
+    else:
+        server_id = "1"
+
+    server_dn = get_server_dn(handle, server_id)
+    return server_dn
+
+
+def _set_adaptor_slot(handle, kwargs):
+    if _is_valid_arg("adaptor_slot", kwargs):
+        adaptor_slot = str(kwargs["adaptor_slot"])
+    else:
+        raise ValueError("Adaptor Slot is not specified as input")
+    return adaptor_slot
+
+
+def _get_adaptor(handle, adaptor_slot, server_id=1, **kwargs):
+
+    server_dn = get_server_dn(handle, server_id)
+    dn = server_dn + "/adaptor-" + str(adaptor_slot)
+    mo = handle.query_dn(dn)
+
+    if mo is None:
+        raise ImcOperationError("Get Adaptor",
+                                "Adaptor is not available")
+
+    return mo
+
+
+def get_vic_adaptor_properties(handle, adaptor_slot, server_id=1, **kwargs):
     """
     This method is used to get the vic adaptor properties
     Args:
         handle (ImcHandle)
-        adaptor_slot (int): PCI slot of the vic adaptor
+        adaptor_slot (string): PCI slot of the vic adaptor
```

Comment:
The adaptor_slot can be a valid string like "slot-L" and hence this has been made into a string.